src/include/OSL/oslexec.h

```diff
@@ -111,6 +111,66 @@ struct SymLocationDesc {
 
 
 
+/// Parameter property hint bitfield values
+enum class ParamHints : uint32_t {
+    /// `none`: No special properties
+    none = 0,
+    /// `interpolated`: This parameter may be an interpolated "user data" or
+    /// "geometric primitive" variable, and so may take on different values at
+    /// different points on the surface or on different objects that share the
+    /// same material. Any optimization of the shader should take this into
+    /// account.
+    interpolated = 1,
+    /// `interactive`: Parameter may have its value interactively modified by
+    /// subsequent ReParameter calls. Any optimization of the shader should
+    /// take this into account.
+    interactive = 2
```

Comment:
Should these enum values be specified in hex so its more obvious they are bitflags vs. a vanilla enum?

Reply:
Considering that the only values now are 1 and 2, I'm not sure it's any clearer in hex. :-)

But I take your point, and I will clarify in the comments about the class itself.

Reply:
C++ has binary literals since C++14 which I think is our floor, right?

Reply:
Yes, 14 is the floor. I've never used binary literals, but they make sense here. Another idiom I've used is

    pepperoni = (1 << 13),
    mushroom  = (1 << 14),
    pineapple = (1 << 15),
    ...

But when the only choices are 1 and 2, I'm not sure any of these language flourishes really improves the clarity. If and when there are enough bits to be confusing, we can choose a different notation.